ui/sv_icons.py

```diff
@@ -0,0 +1,76 @@
+import bpy
+import os
+import glob
+import bpy.utils.previews
+
+DEBUG = False
+
+
+def logDebug(message, extra=""):
+    if DEBUG:
+        print(message, extra)
+
+# custom icon dictionary
+_custom_icons = {}
+
+_custom_icons_loaded = False
+
+
+def custom_icon(name):
+    logDebug("custom_icon called: ", name)
+    global _custom_icons
+    global _custom_icons_loaded
+
+    if not _custom_icons_loaded:
+        load_custom_icons()
+
+    if name in _custom_icons:
+        return _custom_icons[name].icon_id
+    else:
+        logDebug("No custom icon found for name: ", name)
+        return 0
+
+
+def load_custom_icons():
+    logDebug("load_custom_icons called")
+    global _custom_icons
+    global _custom_icons_loaded
+
+    _custom_icons = bpy.utils.previews.new()
+
+    iconsDir = os.path.join(os.path.dirname(__file__), "icons")
+    iconPattern = "sv_*.png"
+    iconPath = os.path.join(iconsDir, iconPattern)
+    iconFiles = [os.path.basename(x) for x in glob.glob(iconPath)]
+    logDebug(iconFiles)
+
+    for iconFile in iconFiles:
+        iconName = os.path.splitext(iconFile)[0]
+        iconID = iconName.upper()
+        logDebug(iconID)
+        _custom_icons.load(iconID, os.path.join(iconsDir, iconFile), "IMAGE")
+
+    _custom_icons_loaded = True
+
+
+def remove_custom_icons():
+    logDebug("remove_custom_icons called")
+    global _custom_icons
+    global _custom_icons_loaded
```

Comment:
This one is more subtle,  we hope to keep the usage of the `global` keyword down. While i understand this is a convenient way to do it, and it will work fine, consider using the property of `_custom_icons` having content to mean that custom_icons _are_ loaded..   `len(_custom_icons_loaded)` will return Truthy whenever it has content, and at no point will the content be half-full . (at least.. i don't see that it would be from your code flow )